internal/binder/binder.go

```diff
@@ -806,7 +806,7 @@ func (b *Binder) bindModuleDeclaration(node *ast.Node) {
 				}
 			}
 			symbol := b.declareSymbolAndAddToSymbolTable(node, ast.SymbolFlagsValueModule, ast.SymbolFlagsValueModuleExcludes)
-			if pattern.StarIndex >= 0 {
+			if pattern.IsValid() && pattern.StarIndex >= 0 {
```

Comment:
I'd prefer if we switched to something like

```go
if pattern := core.TryParsePattern(...); pattern.IsValid() {
  if pattern.StarIndex >= 0 {
    b.file.PatternAmbientModules = append(b.file.PatternAmbientModules, &ast.PatternAmbientModule{Pattern: pattern, Symbol: symbol})
  }
} else {
  b.errorOnFirstToken(name, diagnostics.Pattern_0_can_have_at_most_one_Asterisk_character, name.AsStringLiteral().Text)
}

Reply:
(I can't make the suggestion to above lines, I'm sorry)

Reply:
That code isn't correct, though. `pattern` can be invalid because it contained two stars or because it's the zero struct. The latter case occurs when we don't have a string literal to try to parse with. So that suggestion doesn't work both because we have no argument to pass to `TryParsePattern` and because the error issued is only valid for the double star case.

Reply:
Okay, that specific suggestion might not work, but why not only do this work within `if ast.IsStringLiteral(name)`?

Reply:
https://github.com/andrewbranch/typescript-go/pull/1 is what I had in mind.